Content.Server/Chat/Systems/ChatSystem.cs

```diff
@@ -491,6 +500,26 @@ private void ClientDistanceToList(EntityUid source, int voiceRange, List<ICommon
         }
     }
 
+    private IEnumerable<EntityUid> ListenerEntityDistanceToList(EntityUid source)
+    {
+        if (HasComp<ChatListenerComponent>(source))
+            yield break;
+
+        var transformSource = Transform(source);
+        var sourceMapId = transformSource.MapID;
+        var sourceCoords = transformSource.Coordinates;
+
+        foreach (var (transformComponent, chatListeners) in EntityQuery<TransformComponent, ChatListenerComponent>())
+        {
+            if (source == chatListeners.Owner)
+                continue;
+            if (transformComponent.MapID != sourceMapId || !sourceCoords.InRange(EntityManager, transformComponent.Coordinates, chatListeners.HearingRange))
+                continue;
+
+            yield return chatListeners.Owner;
+        }
```

Comment:
EntityQuery is done in order of components so this is going to iterate every entity in the game internally and be very expensive, use ChatListenerComponent first.